docs/proposals/post-renderer-support.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Support for Post Renderers
+authors:
+  - "@surajnarwade"
+sponsors:
+  - "@anandf"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2024-01-04
+last-updated: 2024-01-04
+---
+
+# Neat Enhancement Idea
+
+Support for Post Renderers
+
+## Summary
+
+This proposal suggests the integration of Kustomize based Helm post renderer support into Argo CD, a feature that will allow users to modify Helm chart manifests at the time of rendering. This enhancement aims to provide greater flexibility in customizing deployments, thereby extending Argo CD's utility in diverse environments.
+
+## Motivation
+
+Helm has a support of post renderers which allows users to manipulate, configure rendered manifest before they are installed. Refer [official documentation](https://helm.sh/docs/topics/advanced/#post-rendering) for more.
+
+[FluxCD](https://fluxcd.io/) already supports the Kustomize based [Helm post renderers](https://fluxcd.io/flux/components/helm/helmreleases/#post-renderers).
+
+This proposal is motivated by fluxCD work and following Github issues on the Argo CD.
+
+### Relavant issues
+
+* https://github.com/argoproj/argo-cd/issues/3698
+* https://github.com/argoproj/argo-cd/issues/7623
+
+### Goals
+
+* Enhance Flexibility: Enable users to apply custom transformations to Helm chart manifests during the deployment process.
+* Increase Compatibility: Broaden the range of Helm charts that can be used with Argo CD, accommodating more complex and specific use cases.
+* Improve User Experience: Provide a more seamless and integrated approach to deploying applications using Argo CD and Helm, enhancing the overall user experience.
+* Encourage Innovation: By allowing more customization, the feature can stimulate more creative and effective deployment strategies within the Argo CD community.
+
+## Proposal
+
+we can add postRenderers field under the Application spec which is very similar solution as fluxCD
+
+```
+  ...
+  source:
+    ...
+    ...
+    helm: 
+      ...
+      postRenderers:
```

Comment:
While the proposal naming comes from Helm's post-renderer feature, this proposal isn't expecting to use it directly, and indeed has a custom implementation for kustomize.

To be more broadly useful could this be under `source.postRenderers` instead, applying to any sort of application source? (e.g. you might want to post-render some external repo container kustomize, or some custom plugin)

Reply:
Agreed. I think `source.kustomizePatches` makes sense.

Whether or not we nest it in a `postRenderers` list is another question.

Reply:
make sense, I have moved the postRenderers under the `source` field now

Reply:
thanks @mikebryant :) @crenshaw-dev I've moved the struct and also linked the POC.